src/Geocoders/CachingGeocoder.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace Maps\Geocoders;
+
+use DataValues\Geo\Values\LatLongValue;
+
+/**
+ * @since 5.0
+ *
+ * @licence GNU GPL v2+
+ * @author HgO < hgo@batato.be >
+ */
+class CachingGeocoder implements Geocoder {
+	protected $geocoder;
+	
+	protected $cache;
+
+	/**
+	 * @param Geocoder $geocoder
+	 * @param BagOStuff $cache
+	 *
+	 */
+	public function __construct( $geocoder, $cache ) {
+		$this->geocoder = $geocoder;
+		$this->cache = $cache;
+	}
+	
+	/**
+	 * @param string $address
+	 *
+	 * @return LatLongValue|null
+	 */
+	public function geocode( $address ) {
+		$coordinates = $this->cache->get( $address );
```

Comment:
I think some kind of namespacing is needed here. Like this you allow users to retrieve whatever cache item they know the keys off, which at best allows them to crash the software by retrieving something that is not a LatLongValue, and at worst allows them to access information they should not have access to.

Reply:
Oh yes, you are right ! It's should be fine now :)